app/assets/javascripts/dateselect.js

```diff
@@ -1,6 +1,28 @@
 $(document).on('ready page:load', function() {
+
+  function enableAllTheseDays(date) {
+    var sdate = $.datepicker.formatDate( 'yy-mm-dd', date)
+    if($.inArray(sdate, gon.available_nights) != -1) {
+        return [true];
+    }
+    return [false];  
+  }
+  
   $('#Check_In').datepicker();
   $('#Check_Out').datepicker();
   $('#First_Night').datepicker();
   $('#Last_Night').datepicker();
+  $('#Couch_Listing_Check_In').datepicker({
+    beforeShowDay: enableAllTheseDays,
+    minDate: 0,
+    numberOfMonths: 3,
+    showButtonPanel: true
+  });
+  $('#Couch_Listing_Check_Out').datepicker({
+    beforeShowDay: enableAllTheseDays,
+    minDate: 0,
+    numberOfMonths: 3,
+    showButtonPanel: true
+  });
```

Comment:
@riverswb this is how many months are being displayed on the little popup when you click it (so right now it shows the next 3 months and their available dates). I'm down to change to 2 or one, 12 might be a little many though to display at once.